KOTOR: Initial save/load game menu implementation

seedhartha opened the pull request:
Accesible from both main menu and ingame options menu.

DrMcCoy commented on src/engines/kotor/gui/saveload.h:
```diff
@@ -0,0 +1,55 @@
+
+namespace KotOR {
+
+enum : uint8 {
```
The typed enum is another C++11 thing we don't want, sorry. :P

seedhartha commented:
Fixed compilation error. Also implemented saved games list loading. There is a bug with listbox item size which I am not yet sure how to workaround. What does DIMENSION field in GUI GFF file stand for?

DrMcCoy commented on src/common/filelist.h:
```diff
@@ -77,6 +77,11 @@ class FileList {
+	/** Add subdirectories of a directory to the list.
+	 *  @param directory directory to scan
+	 */
+	bool addSubDirectories(const UString &directory);
```
Could you separate adding this method into its own commit, please?

DrMcCoy commented on src/engines/aurora/gui.h:
```diff
@@ -75,8 +75,11 @@ class GUI : boost::noncopyable {
+	 * Add a widget.
+	 * @param show show it immediately
+	 */
+	void addWidget(Widget *widget, bool show = false);
```
Changing this method should also be its own commit.

DrMcCoy commented on src/engines/jade/gui/main/main.cpp:
```diff
@@ -57,12 +57,12 @@ MainMenu::MainMenu(Module &module, ::Engines::Console *console) : ::Engines::Kot
+	addWidget(getListBox("ListBoxButtons")->createItem("MINIGAMES"), true);
+	addWidget(getListBox("ListBoxButtons")->createItem("OPTIONS"), true);
+	addWidget(getListBox("ListBoxButtons")->createItem("CREDITS"), true);
+	addWidget(getListBox("ListBoxButtons")->createItem("EXIT"), true);
```
Why? At that point, the GUI itself isn't shown yet.

seedhartha replied:
No idea, that how it worked before :)

DrMcCoy replied:
Huh, am I missing something? You added the parameter `show`, and added `if (show) widget->show()`. So the old behaviour would be the same as `addWidget(..., false);`, no?

seedhartha replied:
In *listbox.cpp*, function *createItem*, there was a non optional call to *show* on newly created item. It didn't work right for me, hence I moved that into *addWidget* function.

DrMcCoy replied:
Ah, okay :)

seedhartha commented:
All done.

seedhartha commented:
Done.

DrMcCoy commented:
Merged as 2c50f7cc...b4ebfdf6, thanks! :)

farmboy0 commented:
What problem do you have with the DIMENSION?
